Sources/SymDiff/source/WholeProgram.cs

```diff
@@ -1470,6 +1482,109 @@ public static List<EquivalenceResult> AllInOneMainWithResult(string[] args)
             p1 = RestructureProgram(p1);
             p2 = RestructureProgram(p2);
 
+            ////////////////////////////////////////////////////////////////////
+            // Structural checks
+            ////////////////////////////////////////////////////////////////////
+            var extractedLoopImplPairs = new Dictionary<string, string>();
+            var p1ImplsToNotExtractLoopsFrom = new HashSet<string>();
+            var p2ImplsToNotExtractLoopsFrom = new HashSet<string>();
+
+            BoogieUtils.ResolveAndTypeCheckThrow(p1, v1name, BoogieUtils.BoogieOptions);
+            BoogieUtils.ResolveAndTypeCheckThrow(p2, v2name, BoogieUtils.BoogieOptions);
+            var p1ImplsWithLoops = p1.Implementations
+                .Where(impl => impl.HasLoops(p1, BoogieUtils.BoogieOptions))
+                .Select(impl => impl.Name)
+                .ToHashSet();
+            
+            if (!Options.noSyntacticCheck)
+            {
+                var (outlinedImpls, structurallyEquivalentImpls) =
+                    StructuralCheckAndOutline(p1, p2,
+                        p1.Implementations.ToHashSet(),
+                        p1.Implementations.Where(impl => p1ImplsWithLoops.Contains(impl.Name)).ToHashSet());
+                structurallyEquivalentImpls.ForEach(p => Options.syntacticEqProcs.Add(
+                    $"{p1Prefix}.{p.Key.Name}", $"{p2Prefix}.{p.Value.Name}"));
+                foreach (var (p1Impl, p2Impl) in structurallyEquivalentImpls)
+                {
+                    p1ImplsToNotExtractLoopsFrom.Add(p1Impl.Name);
+                    p2ImplsToNotExtractLoopsFrom.Add(p2Impl.Name);
+                }
+                ExtendConfiguration(outlinedImpls);
+            }
+
+            // TODO: fix this: outlining seems to break the AST, dump and reparse.
```

Comment:
I think it's worth creating an issue for this.

Reply:
I agree, created #10.